sdk/keyvault/keyvault-keys/src/keyVaultCryptographyClient.ts

```diff
@@ -171,28 +172,25 @@ export class KeyVaultCryptographyClient {
       span.end();
     }
 
-    return { result: result.result!, algorithm, keyID: this.getKeyID() };
+    return {
+      algorithm: encryptParameters.algorithm,
+      result: result.result!,
+      keyID: this.getKeyID(),
+      // TODO: Should we be passing back aad, tag, iv if the service didnt return them but the user passed them in?
+      additionalAuthenticatedData:
```

Comment:
Should we be passing back aad, tag, iv if the service didnt return them but the user passed them in?

Reply:
I believe that for algorithms that don't use `iv` we should throw if the user passes `iv`.

Reply:
These as well for all of the options that only apply for certain cases. We should throw if the user tries to use them for an unexpected algorithm. I believe, at least! I trust your instinct.

Reply:
I did some digging... when I (purposely) allowed `iv` in `encrypt` using A256GCM the service returned a really nice message

```
RestError: An IV should not be in the input for this AES-GCM algorithm. An IV will be generated in the HSM securely and returned. Please make sure to keep it and pass it in when decrypting. (Activity ID: daf37722-7553-11eb-95b5-0242ac120009)
```

So I'd prefer not to validate the service params in this case, it lines up with our guidelines and previous conversations I had on the topic. 

Looking at what dotnet is doing - at least in the `RemoteCryptographyClient` - they are only returning the values that come from the server (except for algorithm it looks like, so that is what I'm inclined to do here too... 🤔 but will wait to see what @heaths recommends.

Reply:
Yes, only return what the service does. And I agree with not validating input for that reason: avoiding tight coupling. That said, .NET and Java, at least, use factory methods to make sure people can only specify valid options given the target invocation, e.g. can't even pass an IV for AES-GCM.

Reply:
Thanks folks! Changed to reflect this conversation.